### PR TITLE
Add querying code from nostr relays

### PR DESCRIPTION
**Pull request by abh3po**

### CODE REVIEW

File: src/scripts/fetcher.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+const FLASH_RELAY = "wss://relay.staging.flashapp.me";
+
+const FlashClient = new NostrClient(FLASH_RELAY, (event: any) => {
+  console.log("GOT EVENT", event);
+});
+
+const handleEvent = function (subscriptionId: string, event: any) {
+  console.log("Handling event", event);
+  delay(1000).then((value) => {
```
Comment (brh28) [inline src/scripts/fetcher.ts]: why the delay?

File: src/scripts/fetcher.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+const relays = [
+  "wss://relay.damus.io",
+  "wss://relay.primal.net",
+  "wss://relay.snort.social",
+];
+
+const FLASH_RELAY = "wss://relay.staging.flashapp.me";
+
+const FlashClient = new NostrClient(FLASH_RELAY, (event: any) => {
```
Comment (brh28) [inline src/scripts/fetcher.ts]: I see .ts files but no types

File: src/scripts/fetcher.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+const fetchEvents = async () => {
+  relays.forEach(async (relay) => {
+    const client = new NostrClient(relay, handleEvent);
+    const subscriptionId = "my-subscription";
+    for (let i = 0; i < 256; i++) {
+      const hexValue = i.toString(16).padStart(2, "0");
+      queryPubkey(hexValue, client, subscriptionId);
+      await delay(5000); // Wait for 5 seconds
```
Comment (brh28) [inline src/scripts/fetcher.ts:49]: another delay?

File: src/scripts/fetcher.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  };
+  client.subscribe(subscriptionId, filters);
+};
+
+const fetchEvents = async () => {
+  relays.forEach(async (relay) => {
+    const client = new NostrClient(relay, handleEvent);
+    const subscriptionId = "my-subscription";
+    for (let i = 0; i < 256; i++) {
+      const hexValue = i.toString(16).padStart(2, "0");
```
Comment (brh28) [inline src/scripts/fetcher.ts:47]: what are these hexValues?

File: src/scripts/fetcher.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  console.log("starting fetch with filter", filter);
+  const filters = {
+    kinds: [0],
+  };
+  client.subscribe(subscriptionId, filters);
+};
+
+const fetchEvents = async () => {
+  relays.forEach(async (relay) => {
+    const client = new NostrClient(relay, handleEvent);
```
Comment (brh28) [inline src/scripts/fetcher.ts]: is everything a 'client'?

**Review by brh28 — COMMENTED** [review]
Since we're still working on the prototype, I'm more concerned with getting something functional rather than focusing on the implementation details, so I didn't leave any direct feedback. Rather, I only left questions that came to mind as I was reading the code (a Socratic review lol). If you do like the code as is, just add some comments to help make it more readable

File: src/scripts/fetcher.ts (hunk 1 above)
Comment (abh3po) [reply src/scripts/fetcher.ts]: can remove this

File: src/scripts/fetcher.ts (hunk 3 above)
Comment (abh3po) [reply src/scripts/fetcher.ts:49]: this one is necessary so we don't flood the relays with queries and they dont have to rate limiting our ips

File: src/scripts/fetcher.ts (hunk 4 above)
Comment (abh3po) [reply src/scripts/fetcher.ts:47]: "00" to "ff" basically first two characters of any pubkey

File: src/scripts/fetcher.ts (hunk 5 above)
Comment (abh3po) [reply src/scripts/fetcher.ts]: I'll name better, maybe "flashClient", & "queryClient"

**Review by brh28 — APPROVED** [review]

**Pull request merged**